### ISSUE TITLE
Tricky definition of samplesperthread in GPU version

### ISSUE THREAD

**Issue by zenandrea** [QMCPACK/qmcpack#2080]
I think there is an ambiguity with the `samplesperthread` variable, at least in the GPU version.

If in the CPU version I want that the DMC has the same number of walkers as those in the VMC, and I set walker for instance to 
`    <parameter name="walkers"> 100 </parameter> `
then I also need to set `samplesperthread` to the same value over $OMP_NUM_THREADS (say 10 in this example):
`    <parameter name="samplesperthread"> 10 </parameter> `
This is in agreement with the manual.

However, if I am running the GPU version, and I want the same as above, then I need to define
`    <parameter name="samplesperthread"> 1 </parameter> `
In other words, in the GPU version `samplesperthread` is indeed the samples per walker!
I think this is a little bit confusing, and not explained in the manual.

By the way, I wonder if the bug with samplesperthread>1 mentioned in #789 has been fixed?
If not, is the GPU version affected by it?

**Comment by prckent** [comment]
It is indeed ambiguous. Unless you can't avoid it for some reason, the best thing to do here is not specify samplesperthread at all. I believe that the CPU OpenMP implementation will do the "right thing" based on the walker count and there is no need to specify anything to do with samples. For optimization only, work in terms of the number of samples. Beware that the CPU OpenMP VMC implementation will adjust the number of walkers to be a multiple of the number of threads so that CPU cores are not wasted.

This is one area where we plan to revise the inputs - most likely with new keywords so that we don't need to be compatible with the logic of the existing code and can easily add error message and abort the application. Send me an email or post here if you have preferences.

**Comment by jtkrogel** [comment]
Settled in batched drivers.
